frontend/src/metabase/admin/settings/components/EmbeddingSettings/StaticEmbeddingSettings.tsx

```diff
@@ -29,14 +32,17 @@ export function StaticEmbeddingSettings({
 
   const isStaticEmbeddingEnabled = useSetting("enable-embedding-static");
 
-  function handleChangeEmbeddingSecretKey(value: string | null) {
-    updateSetting({ key: embeddingSecretKeySetting.key }, value);
-  }
+  const handleChangeEmbeddingSecretKey = async (
+    value: SettingValue<"embedding-secret-key">,
+  ) => await updateSetting({ key: embeddingSecretKeySetting.key }, value);
```

Comment:
nit: I don't see the point of making this async and await the result since we're passing this to `onChange` and we won't use the return result or wait for the the callback to finish before doing other things either.

When I see we await something I'd expect we'd do something after the promise.


```suggestion
  const handleChangeEmbeddingSecretKey = (
    value: SettingValue<"embedding-secret-key">,
  ) => updateSetting({ key: embeddingSecretKeySetting.key }, value);
```